### ISSUE TITLE
Custom Banner Image for Windows Installer

### ISSUE THREAD

**Issue by syxolk** [mp4096/indentex#42]
Now that we have a project logo and icon we can customize the windows installer (hooray)!

It works by adding two images `banner.jpg` and `dialog.jpg` to the package and this two lines in `indentex_template.wxs`:
```
<WixVariable Id="WixUIBannerBmp" Value=".\banner.jpg" />
<WixVariable Id="WixUIDialogBmp" Value=".\dialog.jpg" />
```

It currently looks like this:

![image](https://user-images.githubusercontent.com/6666063/27545408-0e866de8-5a90-11e7-8f53-ab8a7fbe2d1d.png)
![image](https://user-images.githubusercontent.com/6666063/27545441-2055089a-5a90-11e7-9308-41549ff0dae4.png)
![image](https://user-images.githubusercontent.com/6666063/27545466-2df0cd5e-5a90-11e7-80be-590ccb5f2740.png)

Image sizes are [493x58 and 493x312](http://wixtoolset.org/documentation/manual/v3/wixui/wixui_customizations.html).

Images can be JPG or BMP. Windows 8 added support for PNG and other formats but we want to support Windows 7 (?). Since bitmaps are quite large (`dialog.bmp` is 452 KB) I decided to use jpegs with quality 100%.
- https://stackoverflow.com/a/40639894
- https://msdn.microsoft.com/en-us/library/aa367830.aspx

**Comment by mp4096** [comment]
Yep, I like it! :+1: Would you mind opening a PR?

**Comment by syxolk** [comment]
Guide for even smaller images 🤣:
- Load raw image in GIMP
- Click on Image -> Mode -> Indexed...
  - Choose *Generate optimum palette* and *Maximum number of colors*: 256
![image](https://user-images.githubusercontent.com/6666063/27582426-1e19dca8-5b31-11e7-9a2a-f4c0f69cba67.png)
  - Click *Convert*
- Now export the image as bitmap (`*.bmp`)
  - Enable *Run-Length Encoded*
![image](https://user-images.githubusercontent.com/6666063/27582492-619bcaf4-5b31-11e7-8290-73823aa7c05c.png)
  - Click *Export*

Results:

Image | PNG (`optipng -o7`) | JPG (quality=100%) | BMP (palette=256, RLE)
----- | ------ | ----- | -----
banner | 2.7 KB | 7.5 KB | 2.8 KB
dialog | 5.1 KB | 15.1 KB | 6.5 KB

![image](https://user-images.githubusercontent.com/6666063/27582725-475c2d86-5b32-11e7-84a6-b610ef3a9817.png)

**Comment by mp4096** [comment]
Ugh, I have a bad feeling about this. Does WiX work properly with RLE?

**Comment by mp4096** [comment]
And I have to check the colours... I used the full RGB palette.

**Comment by mp4096** [comment]
Ok, the images are so small that I can't see any difference whatsoever.

**Comment by mp4096** [comment]
_*what is it, a banner for ants?*_

**Comment by syxolk** [comment]
> Does WiX work properly with RLE?

I'm pretty sure it works because [someone suggested it in 2008 (!)](https://stackoverflow.com/a/392700):
**.BMP files can be RLE-compressed. I don't know if they support any other compression algorithms.**

>  I used the full RGB palette.

GIMP generates a palette of 256 colors so that it closely resembles the colors of the image. It's *not* a predefined palette.

**Comment by mp4096** [comment]
> GIMP generates a palette of 256 colors

OMG this is crazy. I've completely missed the 'optimal' part. Sure, I have fewer than 256 colours.

**Comment by mp4096** [comment]
Closed in #43
